### ISSUE TITLE
zombie_processes doesn't catch an immediately discarded spawned child

### ISSUE THREAD

**Issue by elee1766** [rust-lang/rust-clippy#17692]
### Summary

so clippy is unable to catch this case in the reproduction, but I feel like it should?

### Lint Name

zombie_processes

### Reproducer

I tried this code:

```rust
fn main() {
    let _ = std::process::Command::new("true").spawn();
    std::thread::sleep(std::time::Duration::from_secs(30));
}
```

I expected to see this happen:

should catch a zombie_process 

Instead, this happened:

no zombie process

### Version

```text
rustc 1.94.1 (e408947bf 2026-03-25)
binary: rustc
commit-hash: e408947bfd200af42db322daf0fadfe7e26d3bd1
commit-date: 2026-03-25
host: x86_64-unknown-linux-gnu
release: 1.94.1
LLVM version: 21.1.8
```
